psycopg/psycopgmodule.c

```diff
@@ -899,6 +914,7 @@ INIT_MODULE(_psycopg)(void)
     /* set some module's parameters */
     PyModule_AddStringConstant(module, "__version__", PSYCOPG_VERSION);
     PyModule_AddStringConstant(module, "__doc__", "psycopg PostgreSQL driver");
+    PyModule_AddIntConstant(module, "__libpq_version__", PG_VERSION_NUM);
```

Comment:
On Mon, Jun 1, 2015 at 6:41 PM, Daniele Varrazzo <notifications@github.com>
wrote:

> In psycopg/psycopgmodule.c
> <https://github.com/psycopg/psycopg2/pull/323#discussion_r31444077>:
>
> > @@ -899,6 +914,7 @@ INIT_MODULE(_psycopg)(void)
> >      /* set some module's parameters */
> >      PyModule_AddStringConstant(module, "__version__", PSYCOPG_VERSION);
> >      PyModule_AddStringConstant(module, "__doc__", "psycopg PostgreSQL driver");
> > +    PyModule_AddIntConstant(module, "__libpq_version__", PG_VERSION_NUM);
>
> I think we may just use strtol(PG_VERSION_HEX, NULL, 0) here and avoid
> introducing PG_VERSION_NUM?
>
I think that is going to be confusing, since both PQserverVersion() and
PQlibVersion() are specified to return the number constructed in decimal,
and we already have a check in skip_before/after_postgres that uses this
exact format.

Moreover, PG_VERSION_HEX is not a string, but also a number like
PG_VERSION_NUM, just constructed differently, so that strtol() call won't
even compile.

Now that I look at how PG_VERSION_HEX is constructed, I think it's going to
be a lot more confusing when PG version number wraps around 10: it's going
to look like, e.g. 0x0a0102, while PG_VERSION_NUM for the same version
would be 100102.  Since we do not even expose the hex version in python, I
would see more sense in moving away from it in favor of decimal one.

--
Alex